plugins/outputs/stackdriver/stackdriver.go

```diff
@@ -0,0 +1,233 @@
+package stackdriver
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"path"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+
+	// Imports the Stackdriver Monitoring client package.
+	monitoring "cloud.google.com/go/monitoring/apiv3"
+	googlepb "github.com/golang/protobuf/ptypes/timestamp"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+// Stackdriver is the Google Stackdriver config info.
+type Stackdriver struct {
+	Project   string
+	Namespace string
+
+	client *monitoring.MetricClient
+}
+
+const (
+	// StartTime for cumulative metrics.
+	StartTime = int64(1)
+	// MaxInt is the max int64 value.
+	MaxInt = int(^uint(0) >> 1)
+)
+
+var sampleConfig = `
+  # GCP Project
+  project = "erudite-bloom-151019"
+
+  # The namespace for the metric descriptor
+  namespace = "telegraf"
+`
+
+// Connect initiates the primary connection to the GCP project.
+func (s *Stackdriver) Connect() error {
+	if s.Project == "" {
+		return fmt.Errorf("Project is a required field for stackdriver output")
+	}
+
+	if s.Namespace == "" {
+		return fmt.Errorf("Namespace is a required field for stackdriver output")
+	}
+
+	if s.client == nil {
+		ctx := context.Background()
+		client, err := monitoring.NewMetricClient(ctx)
+		if err != nil {
+			return err
+		}
+		s.client = client
+	}
+
+	return nil
+}
+
+// Write the metrics to Google Cloud Stackdriver.
+func (s *Stackdriver) Write(metrics []telegraf.Metric) error {
+	ctx := context.Background()
+
+	for _, m := range metrics {
+		for k, v := range m.Fields() {
+			value, err := getStackdriverTypedValue(v)
+			if err != nil {
+				log.Printf("E! Error writing to output [stackdriver]: %s", err)
```

Comment:
Personally, I would change these error strings to describe what failed. Eg: `E! [output.stackdriver] get type failed: %s`

Reply:
Updated now